packages/tinacms/src/auth/TinaCloudProvider.tsx

```diff
@@ -170,6 +175,30 @@ export const TinaCloudProvider = (
     cms.api.tina.setBranch(currentBranch)
   }
 
+  const setupSearch = async (props: CreateClientProps) => {
+    if (cms.api.search) {
+      return
+    }
+    let searchClient
+    // if local and search is configured then we always use the local client
+    // if not local, then determine if search is enabled and use the client from the config
+    if (props.isLocalClient) {
+      searchClient = new LocalSearchClient(cms.api.tina)
+    } else {
+      const hasTinaSearch = Boolean(props.schema.config?.search?.tina)
+      if (hasTinaSearch) {
+        searchClient = new TinaCMSSearchClient(cms.api.tina)
+      } else {
+        searchClient = props.schema.config?.search?.searchClient
+      }
+    }
+
+    if (searchClient) {
+      cms.registerApi('search', searchClient)
+    }
+  }
+  setupSearch(props)
```

Comment:
```suggestion
  React.useEffect(() => {
  setupSearch(props)
  },[]);
```

Reply:
yeah.. I'm not sure why useEffect isn't used in here. I updated this to put the search initialization in a useEffect as suggested: https://github.com/tinacms/tinacms/pull/3949/commits/43e928114e313b3032e44a9ff006bd4462baec1f